kmath-tensors/src/commonMain/kotlin/space/kscience/kmath/tensors/api/TensorAlgebra.kt

```diff
@@ -0,0 +1,252 @@
+     * @return the product of this tensor and [value].
+     */
+    public operator fun TensorStructure<T>.times(value: T): TensorStructure<T>
+
+    /**
+     * Each element of the tensor [other] is multiplied by each element of this tensor.
+     * The resulting tensor is returned.
+     *
+     * @param other tensor to be multiplied.
+     * @return the product of this tensor and [other].
+     */
+    public operator fun TensorStructure<T>.times(other: TensorStructure<T>): TensorStructure<T>
+
+    /**
+     * Multiplies the scalar [value] by each element of this tensor.
+     *
+     * @param value the number to be multiplied by each element of this tensor.
+     */
+    public operator fun TensorStructure<T>.timesAssign(value: T): Unit
+
+    /**
+     * Each element of the tensor [other] is multiplied by each element of this tensor.
+     *
+     * @param other tensor to be multiplied.
+     */
+    public operator fun TensorStructure<T>.timesAssign(other: TensorStructure<T>): Unit
+
+    /**
+     * Numerical negative, element-wise.
+     *
+     * @return tensor negation of the original tensor.
+     */
+    public operator fun TensorStructure<T>.unaryMinus(): TensorStructure<T>
+
+    /**
+     * Returns the tensor at index i
+     * For more information: https://pytorch.org/cppdocs/notes/tensor_indexing.html
+     *
+     * @param i index of the extractable tensor
+     * @return subtensor of the original tensor with index [i]
+     */
+    public operator fun TensorStructure<T>.get(i: Int): TensorStructure<T>
+
+    /**
+     * Returns a tensor that is a transposed version of this tensor. The given dimensions [i] and [j] are swapped.
+     * For more information: https://pytorch.org/docs/stable/generated/torch.transpose.html
+     *
+     * @param i the first dimension to be transposed
+     * @param j the second dimension to be transposed
+     * @return transposed tensor
+     */
+    public fun TensorStructure<T>.transpose(i: Int = -2, j: Int = -1): TensorStructure<T>
+
+    /**
+     * Returns a new tensor with the same data as the self tensor but of a different shape.
+     * The returned tensor shares the same data and must have the same number of elements, but may have a different size
+     * For more information: https://pytorch.org/docs/stable/tensor_view.html
+     *
+     * @param shape the desired size
+     * @return tensor with new shape
+     */
+    public fun TensorStructure<T>.view(shape: IntArray): TensorStructure<T>
+
+    /**
+     * View this tensor as the same size as [other].
+     * ``this.viewAs(other) is equivalent to this.view(other.shape)``.
+     * For more information: https://pytorch.org/cppdocs/notes/tensor_indexing.html
+     *
+     * @param other the result tensor has the same size as other.
+     * @return the result tensor with the same size as other.
+     */
+    public fun TensorStructure<T>.viewAs(other: TensorStructure<T>): TensorStructure<T>
```

Comment:
This operation and the one above it are unsafe and could fail in runtime. Does it make sense to add some level of safety to it by adding annotation or returning nullable value?